File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/report/DailyReservationReport.java
package com.hcmute.reservation.model.dto.report;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class DailyReservationReport {
    private String date;
    private long totalOnline;
    private long totalWalkIn;
    private long total;
    private long noShowCount;
}

File: desktop-staff/reservation-winforms/DTO/report/DailyReservationReport.cs
using Newtonsoft.Json;

namespace reservation_winforms.DTO.report
{
    public class DailyReservationReport
    {
        [JsonProperty("date")]
        public string Date { get; set; } // Chứa "yyyy-MM-dd", "yyyy-MM", hoặc "yyyy"

        [JsonProperty("totalOnline")]
        public long TotalOnline { get; set; }

        [JsonProperty("totalWalkIn")]
        public long TotalWalkIn { get; set; }

        [JsonProperty("total")]
        public long Total { get; set; }

        [JsonProperty("noShowCount")]
        public long NoShowCount { get; set; }
    }
}